apollo-at-link/src/argument.rs

```diff
@@ -0,0 +1,136 @@
+use apollo_compiler::ast::Value;
+use apollo_compiler::schema::{Directive, Name};
+use apollo_compiler::{Node, NodeStr};
+use apollo_federation_error::error::{FederationDirectiveErrorCategory, FederationError};
+use std::ops::Deref;
+
+pub fn directive_optional_enum_argument<'directive>(
+    application: &'directive Node<Directive>,
+    name: &str,
+) -> Option<&'directive Name> {
+    application
+        .arguments
+        .iter()
+        .find(|a| *a.name == *name)
+        .and_then(|a| match a.value.deref() {
+            Value::Enum(name) => Some(name),
+            Value::Null => None,
+            _ => {
+                panic!(
```

Comment:
is this used in tests? We generally avoid panicking, because there'll always be something in production tha triggers it. This could return a `Result<Option<&Name>, SomeErrorType>`

Reply:
It's used to extract arguments from directive applications. The expectation here is that the schema should have been validated by this point, so directive applications should have values of the right types (and if they don't, it's the developer's mistake to have not validated).